File: 771.jewels-and-stones.js
/*
 * @lc app=leetcode id=771 lang=javascript
 *
 * [771] Jewels and Stones
 */

// @lc code=start
/**
 * @param {string} jewels
 * @param {string} stones
 * @return {number}
 */
var numJewelsInStones = function (jewels, stones) {
    let hashMap = {}, ans = 0
    for (const c of jewels) {
        hashMap[c] = 1
    }
    for (const c of stones) {
        if (hashMap[c]) ans++
    }
    return ans
};
// @lc code=end

let jewels = "aA", stones = "aAAbbbb";
let answer = numJewelsInStones(jewels, stones);
console.log(answer)

File: 771.jewels-and-stones.cpp
#include <iostream>
#include <algorithm>
using namespace std;

/*
 * @lc app=leetcode id=771 lang=cpp
 *
 * [771] Jewels and Stones
 */

// @lc code=start
class Solution
{
public:
    int numJewelsInStones(string jewels, string stones)
    {
        int num = 0;
        for (int i = 0; i < jewels.size(); i++)
        {
            num += count(stones.begin(), stones.end(), jewels[i]);
        }
        return num;
    }
};
// @lc code=end
int main()
{
    Solution ob;

    string jewels = "aA", stones = "aAAbbbb";

    int answer = ob.numJewelsInStones(jewels, stones);
    cout << answer << endl;
}
